src/lib/swapper/LifiSwapper/executeTrade/executeTrade.ts

```diff
@@ -0,0 +1,152 @@
+import type Lifi from '@lifi/sdk/dist/Lifi'
+import type { Route, Step } from '@lifi/sdk/dist/types'
+import type { BuildSendTxInput, EvmChainAdapter, EvmChainId } from '@shapeshiftoss/chain-adapters'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { TradeResult } from '@shapeshiftoss/swapper'
+import { SwapError, SwapErrorType } from '@shapeshiftoss/swapper'
+import type { Signer } from 'ethers'
+import { providers } from 'ethers'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { SELECTED_ROUTE_INDEX } from 'lib/swapper/LifiSwapper/utils/constants'
+import { getLifi } from 'lib/swapper/LifiSwapper/utils/getLifi'
+import type { LifiExecuteTradeInput } from 'lib/swapper/LifiSwapper/utils/types'
+
+const createBuildSendTxInput = async (
+  lifi: Lifi,
+  wallet: HDWallet,
+  accountNumber: number,
+  lifiStep: Step,
+): Promise<BuildSendTxInput<EvmChainId>> => {
+  const transactionRequest: providers.TransactionRequest = await (async () => {
+    const transactionRequest = lifiStep?.transactionRequest
+
+    if (transactionRequest !== undefined) return transactionRequest
+
+    // if transactionRequest is not present, request it
+    const { transactionRequest: newTransactionRequest } = await lifi.getStepTransaction(lifiStep)
+    return newTransactionRequest ?? {}
+  })()
+
+  const { value, to, gasPrice, gasLimit, data } = transactionRequest
+
+  if (
+    transactionRequest === undefined ||
+    value === undefined ||
+    to === undefined ||
+    gasPrice === undefined ||
+    gasLimit === undefined ||
+    data === undefined
+  ) {
+    throw new SwapError('[executeTrade] - incomplete or undefined transaction request', {
+      code: SwapErrorType.EXECUTE_TRADE_FAILED,
+      details: { transactionRequest },
+    })
+  }
+
+  return {
+    value: value.toString(),
+    wallet,
+    to,
+    chainSpecific: {
+      gasPrice: gasPrice.toString(),
+      gasLimit: gasLimit.toString(),
+      maxFeePerGas: undefined,
+      maxPriorityFeePerGas: undefined,
+    },
+    accountNumber,
+    memo: data.toString(),
+  }
+}
+
+export const executeTrade = async ({
+  trade,
+  wallet,
+}: LifiExecuteTradeInput): Promise<TradeResult> => {
+  try {
+    const lifi = getLifi()
+
+    const { accountNumber, sellAsset, routesRequest } = trade
+
+    if (window.ethereum === undefined) {
+      throw new SwapError('[executeTrade] - window.ethereum is undefined', {
+        code: SwapErrorType.EXECUTE_TRADE_FAILED,
+      })
+    }
```

Comment:
Not sure if that's here as part of the draft for debugging purposes, but we should not assume `window.ethereum`, as not all wallet we support / supported wallets as part of the li.fi feature are injected with `window.ethereum`.

Native and KeepKey are not EIP-1193 providers and work in a very different way, so that would effectively fail for these
```suggestion
```